### PR TITLE
Feature/sum of scores

### PR DESCRIPTION
**Pull request by noahdurbin**
## Type Of Change
- [X] New Feature
- [ ] Debugging
- [X] Refactor/File Clean-up
- [X] Tests
- [ ] Documentation

## Provide A Summary Of Your Change

Refactored the game factory to use a class variable that stores all objects created using the method creating games from CSV data. Also added methods for calculating the highest scoring game and lowest scoring game along with helper methods.

## Group Project Check List
- [X] I made test for my code, and made sure they were passing before submitting PR
- [X] I assigned at least 2 group members to review my PR

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -1,21 +1,40 @@
...
+    @date_time = game_data[:date_time].to_i
+    @away_team_id = game_data[:away_team_id].to_i
+    @home_team_id = game_data[:home_team_id].to_i
+    @away_goals = game_data[:away_goals].to_i
+    @home_goals = game_data[:home_goals].to_i
     @venue = game_data[:venue]
     @venue_link = game_data[:venue_link]
   end
+
+  def self.all_games
```
Comment (zach-bergman) [inline lib/game.rb:21]: Nice I like this method, instead of having to call GameFactory.all_games a bunch

File: spec/game_spec.rb [hunk 2]
```diff
@@ -23,4 +23,35 @@
...
+
+      expect(Game.all_games).to be_a(Array)
+      expect(Game.all_games.length).to eq(20)
+      expect(Game.all_games).to all be_a(Game)
+    end
+
+    it 'can calculate the total score of a game' do
+      game1 = Game.new({game_id: 2012030221, season: 20122013, type: :Postseason, date_time: 5/16/13, away_team_id: 3, home_team_id: 6, away_goals: 2, home_goals: 3, venue: 'Toyota Stadium', venue_link: '/api/v1/venues/null'})
+
+      expect(game1.total_score).to eq(5)
```
Comment (zach-bergman) [inline spec/game_spec.rb:40]: Great test for the helper

**Review by zach-bergman — APPROVED** [review]
Looks good! First couple of methods out of the way

File: lib/game.rb (hunk 1 above)
Comment (austincarrjones) [reply lib/game.rb:21]: yeah nice abstraction!

**Review by austincarrjones — APPROVED** [review]
nice work on these first methods! very readable

**Pull request merged**